bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2BridgeHandler.java

```diff
@@ -0,0 +1,667 @@
+            task.cancel(false);
+        }
+        int milliSeconds;
+        if (retryApplicationKey) {
+            // short delay used during attempts to create or validate an application key
+            milliSeconds = FAST_SCHEDULE_MILLI_SECONDS;
+            applKeyRetriesRemaining--;
+        } else {
+            // default delay, set via configuration parameter, used as heart-beat 'just-in-case'
+            Clip2BridgeConfig config = getConfigAs(Clip2BridgeConfig.class);
+            milliSeconds = config.checkMinutes * 60000;
+            if (retryConnection) {
+                // exponential back off delay used during attempts to reconnect
+                int backOffDelay = 60000 * (int) Math.pow(2, RECONNECT_MAX_TRIES - connectRetriesRemaining);
+                milliSeconds = Math.min(milliSeconds, backOffDelay);
+                connectRetriesRemaining--;
+            }
+        }
+        checkConnectionTask = scheduler.schedule(() -> checkConnection(), milliSeconds, TimeUnit.MILLISECONDS);
+    }
+
+    /**
+     * If a child thing has been added, and the bridge is online, update the child's data.
+     */
+    @Override
+    public void childHandlerInitialized(ThingHandler childHandler, Thing childThing) {
+        if (thing.getStatus() == ThingStatus.ONLINE && (childHandler instanceof Clip2ThingHandler)) {
+            logger.debug("childHandlerInitialized() {}", childThing.getUID());
+            try {
+                ResourceReference reference = ((Clip2ThingHandler) childHandler).getResourceReference();
+                getClip2Bridge().getResources(reference).getResources().forEach(r -> onResource(r));
+            } catch (ApiException | AssetNotLoadedException e) {
+                // exceptions should not occur here; but log anyway (just in case)
+                logger.warn("childHandlerInitialized() {}", e.getMessage(), e);
+            }
+        }
+    }
+
+    /**
+     * Return the application key for the console app.
+     *
+     * @return the application key.
+     */
+    public @Nullable String consoleGetApplicationKey() {
+        Clip2BridgeConfig config = getConfigAs(Clip2BridgeConfig.class);
+        return config.applicationKey;
+    }
+
+    /**
+     * Return the ip address for the console app.
+     *
+     * @return the ip address.
+     */
+    public @Nullable String consoleGetIpAddress() {
+        Clip2BridgeConfig config = getConfigAs(Clip2BridgeConfig.class);
+        return config.ipAddress;
+    }
+
+    /**
+     * Return a list of resources for the console app.
+     *
+     * @param reference the resource reference to return.
+     * @return list of resources of the given type.
+     */
+    public List<Resource> consoleGetResources(ResourceReference reference) {
+        try {
+            return getClip2Bridge().getResources(reference).getResources();
+        } catch (ApiException | AssetNotLoadedException e) {
+        }
+        return List.of();
+    }
```

Comment:
This one actually isn't needed at all, since it duplicates the method below. The console consumer can do the exception -> empty list conversion by itself.

Reply:
> This one actually isn't needed at all, since it duplicates the method below.

It does not duplicate the other method. The console command is used to detect a hue bridge **_that has not yet been created as a thing_** and so that it can eventually be instantiated as a thing, and read its contents so that they can also be eventually instantiated as things. The other method requires that the bridge thing will already have been initialized and online. i.e. the console command is to help users set up the binding initially..

Reply:
It does. The contents (`getClip2Bridge().getResources(reference).getResources()`) are exactly the same, the only difference is a logging statement and the exception handling. That's why I said this method can be removed when the exception handling is moved to the caller.

Reply:
@maniac103 what makes you so sure that the two methods are the same? The former calls `checkAssetsLoaded()` and the latter does not..

```
    public Resources getResources(ResourceReference reference) throws ApiException, AssetNotLoadedException {
        logger.debug("getResources() {}", reference);
        checkAssetsLoaded();
        return getClip2Bridge().getResources(reference);
    }

    public List<Resource> getResourcesConsole(ResourceReference reference) {
        try {
            return getClip2Bridge().getResources(reference).getResources();
        } catch (ApiException | AssetNotLoadedException e) {
        }
        return List.of();
    }
```

Reply:
^
But, you know what, it still works if the console does call `checkAssetsLoaded()`. So I will adopt your request.